### PR TITLE
change yiq mixin to an actual function

### PR DESCRIPTION
**Pull request by gijsbotje**
to me it seemed weird that this is a mixin that always returns the line `color: #xxx;` 
i modified it to be an actual function and only return a value to be used in any property

also updated every use of the 'function' to use the actual function

### CODE REVIEW

**Comment by chrispatterson** [comment]
This is definitely something that has tripped me up - like the changes.

**Comment by XhmikosR** [comment]
There's also #23748, #23191 and #22996. Not sure what's the best.

@mdo

**Review by mdo — APPROVED** [review]

**Comment by gijsbotje** [comment]
@mdo are the hound-ci errors left really neccessary to be changed?

**Comment by XhmikosR** [comment]
Please ignore Hound until https://github.com/twbs/bootstrap/pull/24221 is merged and revert the patch.

**Comment by XhmikosR** [comment]
@gijsbotje: please revert your last patches.

@mdo: did you see https://github.com/twbs/bootstrap/pull/23943#issuecomment-332442550?

**Comment by gijsbotje** [comment]
@XhmikosR travis says
`Expected "color" to come before "text-decoration"                order/properties-order`

**Comment by XhmikosR** [comment]
@gijsbotje: that is right. You should address any Travis errors.

File: scss/mixins/_badge.scss [hunk 1]
```diff
@@ -1,11 +1,11 @@
 @mixin badge-variant($bg) {
-  @include color-yiq($bg);
+  color: color-yiq($bg);
   background-color: $bg;
 
   &[href] {
     @include hover-focus {
-      @include color-yiq($bg);
       text-decoration: none;
+      color: color-yiq($bg);
```
Comment (XhmikosR) [inline scss/mixins/_badge.scss]: Here's your error @gijsbotje

**Comment by gijsbotje** [comment]
@XhmikosR all done :)

**Comment by XhmikosR** [comment]
Cool, waiting for @mdo to choose which one of those related PRs we should merge, see https://github.com/twbs/bootstrap/pull/23943#issuecomment-332442550.

**Pull request merged**